src/Illuminate/Database/Grammar.php

```diff
@@ -116,6 +116,62 @@ protected function wrapValue($value)
         return $value;
     }
 
+    /**
+     * Convert array into sql group.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function group($values)
+    {
+        return '('.implode(', ', $values).')';
+    }
+
+    /**
+     * Recursively group arrays containing values into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupify($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
+        }
+
+        return $this->parameter($values);
+    }
+
+    /**
+     * Recursively group arrays containing columns into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupifyColumns($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
```

Comment:
Is there any valid sql query where recursion is required? I would probably just map the `wrap` method to it.

Reply:
Same here. Except instead of needing to call columnize we can just call wrap around the values since when they bubble up they will then be grouped correctly.

Reply:
Ok looks like `select * from table where (colA, (colB, colC)) in ((valA, (valB, valC)))` is a valid syntax (at least under MySQL(. I assumed it breaks the query. So this is good I guess.